File: Java/Actividad1/src/ejercicio26/sala.java
package ejercicio26;

public class sala {
    //aqui creo mi matris del objeto
    private asiento[][] asientos;
    //constructor, creo la matriz vacia
    public sala(int filas, int columnas) {
        asientos = new asiento[filas][columnas];

        //el respectivo recorrido de la matriz
        for (int i = 0; i < filas; i++) {
            for (int j = 0; j < columnas; j++) {
                asientos[i][j] = new asiento(i, j);
            }
        }
    }
    // el mostrar sala, use el for each
    public void mostrarSala() {
        for (asiento[] fila : asientos) {
            for (asiento a : fila) {
                System.out.print(a.mostrar());
            }
            System.out.println();
        }
    }
    //el reservar y liberar, llamo los metodos creado en mi objeto
    public void reservar(int fila, int columna) {
        asientos[fila][columna].reservar();
    }

    public void liberar(int fila, int columna) {
        asientos[fila][columna].liberar();
    }
}

File: CSharp/Actividad1/Actividad1/ejercicio26/Sala.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Actividad1.ejercicio26
{
    public class Sala
    {
        //aqui a diferencia de java se usa la matriz de objeto
        private Asiento[,] asientos;

        public Sala(int filas, int columnas)
        {
            asientos = new Asiento[filas, columnas];

            for (int i = 0; i < filas; i++)
            {
                for (int j = 0; j < columnas; j++)
                {
                    asientos[i, j] = new Asiento(i, j);
                }
            }
        }
        public void Mostrar()
        {
            for (int i = 0; i < asientos.GetLength(0); i++)
            {
                for (int j = 0; j < asientos.GetLength(1); j++)
                {
                    Console.Write(asientos[i, j].Mostrar());
                }
                Console.WriteLine();
            }
        }
        public void Reservar(int f, int c)
        {
            asientos[f, c].Reservado = true;
        }
        public void Liberar(int f, int c)
        {
            asientos[f, c].Reservado = false;
        }

    }
}
